### ISSUE TITLE
Batchnorm kernel compilation fail

### ISSUE THREAD

**Issue by avidov** [ROCmSoftwarePlatform/MIOpen#796]
I am working to add MiOpen backend to my product. I'm testing on AWS g4ad.16xlarge with AMD Radeon Pro V520 GPUs and Ubuntu 18.04.
I tried both installing MIOpen using apt-get (https://rocmdocs.amd.com/en/latest/Installation_Guide/Installation-Guide.html) and from source. In both cases the BatchNorm kernel compilation fails when calling miopenBatchNormalizationForwardTraining() while miopenBatchNormalizationForwardInference() works.

The error log below is from running `make check` from MiOpen source build. 

10/57 Test  #3: test_bn_3d_spatial_test ..........***Failed  Error regular expression found in output. Regex=[FAILED]  0.70 sec
Choosing smaller input values for low dims
<inline asm>:14:23: error: not a valid operand.
v_add_f32 v26 v26 v26 row_bcast:15 row_mask:0xa
                      ^
<inline asm>:15:23: error: not a valid operand.
v_add_f32 v25 v25 v25 row_bcast:15 row_mask:0xa
                      ^
<inline asm>:17:23: error: not a valid operand.
v_add_f32 v26 v26 v26 row_bcast:31 row_mask:0xc
                      ^
<inline asm>:18:23: error: not a valid operand.
v_add_f32 v25 v25 v25 row_bcast:31 row_mask:0xc
                      ^
/home/avidov/ROCm/MIOpen/build/bin/test_bn_3d_spatial_test --float --input 16, 32, 8, 8, 8 
FAILED: /home/avidov/ROCm/MIOpen/src/ocl/clhelper.cpp:119: Error Building OpenCL Program in BuildProgram()
error: cannot compile inline asm
error: cannot compile inline asm
error: cannot compile inline asm
error: cannot compile inline asm
4 errors generated.
Error: Creating the executable from LLVM IRs failed.
 Build Program Failure
Forward Train Spatial Batch Normalization: 
Input tensor: 16, 32, 8, 8, 8
<inline asm>:14:23: error: not a valid operand.
v_add_f32 v26 v26 v26 row_bcast:15 row_mask:0xa
                      ^
<inline asm>:15:23: error: not a valid operand.
v_add_f32 v25 v25 v25 row_bcast:15 row_mask:0xa
                      ^
<inline asm>:17:23: error: not a valid operand.
v_add_f32 v26 v26 v26 row_bcast:31 row_mask:0xc
                      ^
<inline asm>:18:23: error: not a valid operand.
v_add_f32 v25 v25 v25 row_bcast:31 row_mask:0xc

**Comment by atamazov** [comment]
Please _attach_ log taken with `export MIOPEN_LOG_LEVEL=6`.

**Comment by avidov** [comment]
Thank you for the reply. The log is attached.

[make-check.log](https://github.com/ROCmSoftwarePlatform/MIOpen/files/6118982/make-check.log)

**Comment by atamazov** [comment]
Your GPU is gfx1011. It is not fully supported in 2.9. You can try to filter our assembly kernels, see https://github.com/ROCmSoftwarePlatform/MIOpen/blob/develop/doc/src/DebugAndLogging.md#filtering-by-build-method.

**Comment by avidov** [comment]
Tries the following with similar result
`export MIOPEN_DEBUG_GCN_ASM_KERNELS=0`
`export MIOPEN_DEBUG_GCN_ASM_KERNELS=1`
`export MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=0`
`export MIOPEN_DEBUG_AMD_ROCM_PRECOMPILED_BINARIES=1`

Although, from the log it looks like the process tries to build  'MIOpenBatchNormFwdTrainSpatial.cl' from source rather than using assembly or precomplied one. 
`FAILED: /home/avidov/ROCm/MIOpen/src/ocl/clhelper.cpp:119: Error Building OpenCL Program in BuildProgram()`

**Comment by atamazov** [comment]
It looks like gfx10 GPUs are not supported in the latest ROCm release https://github.com/RadeonOpenCompute/ROCm#supported-gpus. Please expect gfx10 support in the upcoming releases.

**Comment by avidov** [comment]
I see.
Thank you a bunch for the great and quick support Artem !
